### PR TITLE
gdal 3.1.3

### PR DESCRIPTION
**Pull request by Moisan**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by miccal** [comment]
I think the three patches I added to `gdal` when I [updated `poppler`](https://github.com/Homebrew/homebrew-core/pull/60513) can be removed.

**Comment by miccal** [comment]
https://github.com/OSGeo/gdal/pull/2931 perhaps?

**Comment by Moisan** [comment]
It fails even with OSGeo/gdal#2931 (I tested locally).

**Comment by fxcoudert** [comment]
I've opened what I hope is a clear bug report about the issue: https://github.com/OSGeo/gdal/issues/3012

**Comment by miccal** [comment]
I have successfully built and tested this on my 10.14 machine by adding:
```
inreplace "frmts/jpeg2000/jpeg2000_vsil_io.cpp", "static int JPEG2000_VSIL_write(jas_stream_obj_t *obj, char *buf, unsigned int cnt)", "static int JPEG2000_VSIL_write(jas_stream_obj_t *obj, const char *buf, unsigned int cnt)"
```
between `def install` and `args = [`.

I got the idea from [here](https://raw.githubusercontent.com/jrmarino/Ravenports/master/bucket_9F/gdal).

I hope that helps.

**Comment by fxcoudert** [comment]
As a note: upstream recommends removing jasper and depending on openjpeg instead. Quoting: “it's a better alternative”. We might consider that advice.

**Comment by fxcoudert** [comment]
Upstream fix committed in https://github.com/OSGeo/gdal/commit/9ef8e16e27c5fc4c491debe50bf2b7f3e94ed334

File: Formula/gdal.rb [hunk 1]
```diff
@@ -119,7 +97,7 @@ def install
       "--with-cfitsio=#{Formula["cfitsio"].opt_prefix}",
       "--with-hdf5=#{Formula["hdf5"].opt_prefix}",
       "--with-netcdf=#{Formula["netcdf"].opt_prefix}",
-      "--with-jasper=#{Formula["jasper"].opt_prefix}",
+      "--with-openjpeg",
```
Comment (SMillerDev) [inline Formula/gdal.rb:100]: We don't need to specify the location here?

Comment (fxcoudert) [reply Formula/gdal.rb:100]: No, the `--with-openjpeg` flag does not accept any argument

**Comment by seisman** [comment]
It seems that the latest GDAL doesn't support JPEG2000 format. I'm not sure if it's a GDALv3.1.3 bug, or it's because this PR changes jasper to openjpeg.

To reproduce the issue, you can run:
```
$ wget --no-check-certificate https://oceania.generic-mapping-tools.org/server/earth/earth_relief/earth_relief_15s_p/N00E000.earth_relief_15s_p.jp2
$ gdalinfo N00E000.earth_relief_15s_p.jp2                                     
ERROR 4: `N00E000.earth_relief_15s_p.jp2' not recognized as a supported file format.
gdalinfo failed - unable to open 'N00E000.earth_relief_15s_p.jp2'.
```

**Comment by fxcoudert** [comment]
@seisman I've asked the upstream maintainer at https://github.com/OSGeo/gdal/issues/3012

**Comment by fxcoudert** [comment]
@seisman can you check the output of `gdalinfo --format JP2OpenJPEG`?

**Comment by seisman** [comment]
@fxcoudert Thanks for your quick fix. I can confirm that it works well now.

**Pull request closed without merging**